osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -916,18 +916,24 @@ private bool transferRulesetValue()
             return true;
         }
 
-        private void delete(BeatmapSetInfo? beatmap)
+        /// <summary>
+        /// Request to delete the current beatmap.
+        /// </summary>
+        public void DeleteBeatmap()
```

Comment:
Was there a particular reason to change these methods to always use `Beatmap.Value` rather than explicitly take in a `Beatmap{Set}Info`? I don't particularly like this change if I'm honest... (rationale being that it's more magic and less explicit).

Reply:
It's just to write one less dependency in BDL, and hope that it's only ever used for the current beatmap/set.

Edit also seems to have an optional param, and it's only ever used in tests (and the real usage is just passing the current beatmap):
https://github.com/ppy/osu/blob/491f15e68ced2be97736445775c62eeccfae60ca/osu.Game/Screens/Select/SongSelect.cs#L375

Reply:
Yeah I dunno, I'm not feeling the tradeoff there. I'd rather have the extra BDL dependency.

Reply:
I think this needs to be restored, because i've made the popover specific to the beatmap at the time the popover was opened (this needs to be respected).

I'll leave it as a follow-up task for you @Joehuu, as I was focused on getting the UI/UX to feel more correct.

Reply:
I don't think it's correct UX to leave the popover with the last beatmap after I randomed (F2) out of it. I would just hide the popover when the beatmap changes.

As for just code quality, I'll revert this to keep things moving.

Reply:
For reference we were briefly considering preventing changes to beatmap selection when the popover is open in twitch stream chat. No strong opinions from me which one is preferable though,

Reply:
I did the hide as it was easier.

The other proposal may require dimming the whole screen to indicate you can't navigate.